### DISCUSSION TITLE
Stop object does not stop movement of Pathfinding movements

### DISCUSSION THREAD

**Discussion by shinobirozs** [4ian/GDevelop#2328] [Enhancement ideas (and their technical discussions)]
## Describe the bug
There seems to be no way to stop a moving object with the Pathfinding behaviour.

## To Reproduce
Steps to reproduce the behaviour:
1. Add a Pathfinding behaviour to an object
2. Move the object to the mouse position
3. Use Stop object action to stop the moving object
4. The object doesn't stop moving.

## Other details
* Mac
* The desktop app V 5.0.0-beta75

Thank you.

**Comment by 4ian** [comment]
As a work around, you can try calling the action to go to a position and set the position to the X and Y position of the object.

**Comment by Wend1go** [comment]
I think the ideal behaviour would be to stop the moving object in place when the "Pathfinding Behavior" is disabled and continue moving when it is enabled again.
After a quick glance on https://github.com/4ian/GDevelop/blob/master/Extensions/PathfindingBehavior/pathfindingruntimebehavior.js I don't even see an `onDeactivated` function. :confused:

**Comment by 4ian** [comment]
When a behavior is disabled, `doStepPreEvents` won't be called so, while I've not tested this recently, this should still work (i.e: disabling the behavior should disable the movement). Movement will still resume as soon as you enable again the behavior.

**Comment by shinobirozs** [comment]
Thank you. I'll try using the workaround first. It would be great if we can use Stop object action in the future since it was the first action that comes to mind if we want to stop object movements. 

Cheers!

**Comment by 4ian** [comment]
Add this on the roadmap: https://trello.com/c/ABOAzUjj/326-add-action-for-stopping-objects-using-the-pathfinding-behavior

**Comment by shinobirozs** [comment]
Hi,

I tried the work around by moving the character to the character's current position but it teleports it back to the original position as seen below. The workaround seems to not solve the issue. Thanks.

![ezgif com-optimize](https://user-images.githubusercontent.com/29979672/60511010-839bcf80-9d03-11e9-88af-639025426e42.gif)

![image](https://user-images.githubusercontent.com/29979672/60511100-bfcf3000-9d03-11e9-94cb-cddbfd66f95a.png)

**Comment by shinobirozs** [comment]
I tried to disable the Pathfinding behavior and it did stop the characters movement. I'll use this as a workaround for now. Thanks for your time.

![image](https://user-images.githubusercontent.com/29979672/60511626-08d3b400-9d05-11e9-8272-3b130f26d634.png)

**Comment by 4ian** [comment]
Great that it's working with this - nice graphics by the way!

>  but it teleports it back to the original position as seen below

Indeed, that's a nasty bug! I fixed this for next version :)
Can confirm that setting the destination to the object X;Y position will properly stop the object.
